linode/manager
# feat: [M3-6968] Add DC specific pricing info to migration modal

## Description 📝
Adds a price comparison when trying to migrate a linode and selecting a region with an DC increase.

Per the mockups, we're looking at aligning the old region a new region data in a row, as well as an optional pricing dataset underneath if the migration is set towards a DC with a price increase

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-08-23 at 00 14 56](https://github.com/linode/manager/assets/130582365/7b987037-4358-496b-a174-470183141f27) | ![Screenshot 2023-08-23 at 00 14 25](https://github.com/linode/manager/assets/130582365/45db794a-25cb-420a-b091-56e4b5e7b484) |

## How to test 🧪
1. Pull local code
2. Navigate to the linodes landing page or an individual linode
3. Use the action menu and click on "Migrate"
4. Play with the UI and confirm the test scenari outlined in the test file:
- should render the initial ConfigureForm fields on modal open (no MigrationPricing)
- shouldn't render the MigrationPricing component when the current Linode region is selected in the "New region" select
- shouldn't render the MigrationPricing component when a region without price increase is selected in the "New region" select (anything else than `br-gru` & `id-cgk`)
- should render the MigrationPricing component when a region with price increase is selected (ex: `Sao Paulo br-gru`)
- should render the MigrationPricing component when a region with price decrease is selected (ex: `Sao Paulo br-gru`)
- shouldn't render the MigrationPricingComponent for any case if the flag is disabled
